### ISSUE TITLE
(Speech recognition) hope to get inside sMBR, MWER tensorflow code

### ISSUE THREAD

**Issue by liangzhenlin123** [tensorflow/addons#415]
**System information**
- TensorFlow version (you are using): 1.12
- TensorFlow Addons version: Not used yet
- Is it in the tf.contrib (if so, where):No
- Are you willing to contribute it (yes/no):Yes
- Are you willing to maintain it going forward? (yes/no):yes

**Describe the feature and the current behavior/state.**
Smbr and mwer are important loss functions for speech recognition, and google has published many papers based on these two losses, I hope to get open source code.

**Will this change the current api? How?**

**Who will benefit with this feature?**
All speech recognition researchers
**Any Other info.**

**Comment by vijaysumaravi** [comment]
How do I +1 to this?

**Comment by facaiy** [comment]
@liangzhenlin123 Hi, would you mind listing the original papers? and if possible, its open source reference implementation in other framework (say, pytorch)

**Comment by utkarshsimha** [comment]
Would love to work on this - given more information! 
Here are some resuources I found for them: 
[sMBR](https://arxiv.org/pdf/1706.02776.pdf)
[MWER](https://arxiv.org/pdf/1712.01818.pdf)

**Comment by facaiy** [comment]
Welcome, Utkarsh! Feel free to reach out us if you need any help :-)

**Comment by prakharcode** [comment]
Hi, @utkarshsimha are you taking this? I would like to pick it up if this is open. @facaiy

**Comment by facaiy** [comment]
Welcome, Prakhar!

@utkarshsimha Hi, Utkarsh, could you keep us updated? Thanks?

**Comment by utkarshsimha** [comment]
@facaiy @prakharcode - Sorry for the delayed reply.
I have been reading the paper and understanding the mechanics. 

Prakhar, if you would like to collaborate, I would be open to that! :) 

Thanks!

**Comment by liangzhenlin123** [comment]
@facaiy @utkarshsimha @prakharcode sorry for the delayed reply :(  I have received inspiration elsewhere, and I will finished the N-best MWER in two days, maybe include the sample MWER though it is more difficult than N-best....

**Comment by TeaPoly** [comment]
> @facaiy @utkarshsimha @prakharcode sorry for the delayed reply :( I have received inspiration elsewhere, and I will finished the N-best MWER in two days, maybe include the sample MWER though it is more difficult than N-best....

Can you share MWER for us, or tell us where are your inspiration.

**Comment by TeaPoly** [comment]
> **System information**
> 
> * TensorFlow version (you are using): 1.12
> * TensorFlow Addons version: Not used yet
> * Is it in the tf.contrib (if so, where):No
> * Are you willing to contribute it (yes/no):Yes
> * Are you willing to maintain it going forward? (yes/no):yes
> 
> **Describe the feature and the current behavior/state.**
> Smbr and mwer are important loss functions for speech recognition, and google has published many papers based on these two losses, I hope to get open source code.
> 
> **Will this change the current api? How?**
> 
> **Who will benefit with this feature?**
> All speech recognition researchers
> **Any Other info.**

There is a repo which may be helpeful to you: https://github.com/datemoon/tf-code-acoustics. There are some SDT methods.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
